BeeStation/BeeStation-Hornet
# Server framework v2

## About The Pull Request
upgrades tcomms and RnD servers with thermal mechanics so they need to be cooled, generate heat and stop working if overheated.
tcomms monitoring console now lets you monitor the temperatures of tcomms machines. uses NTnet.
makes RnD servers upgrades make the servers use less power (and therefore less heat) 
both generate heat based on power used

- [x] Integrate RnD servers
- [x] Spark when overheated
- [x] TGUIfy tcomms monitoring console
- [x] NTnetify tcomms

~~Requires https://github.com/BeeStation/BeeStation-Hornet/pull/10964 so tcomms don't overheat roundstart~~ merged


## Why It's Good For The Game
more ways to non destructively sabotage tcomms
more responsive and working rnd servers
easier to add more server like machines 


## Testing Photographs and Procedure
tcomms:
- go to tcomms
- check telecommunications monitoring console
- all tcomms working fine
- turn off the freezer
- tcomms temperature rising and slowly overheating
- say something on radio
- theres lag
- when tcomms fully overheats it turns off until cooled down and sparks

RnD servers:
- check roundstart servers
- working fine
- add more servers without upgrading anything
- they begin to overheat
- upgrade the servers and/or freezer
- servers cool down depending on upgrades


<details>
<summary>Screenshots&Videos</summary>

All tcomms working fine

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/8c45a632-5ebf-48f8-bd20-e6da7c29608e)

tcomms overheating without cooling

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/ef8e731c-705d-4ee0-af53-2953cfd0ed8e)


tcomms overheated

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/5f50a352-7362-48f2-bb49-914986a5bb46)

roundstart RnD servers working fine

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/4a0b6b9c-f7eb-4d77-b155-fb624e76dffc)

adding more RnD servers

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/e9bae993-a89f-4aa5-aac9-81d913be0288)

![image](https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/63145f50-03e1-4d99-845a-cc0bcc19ffd1)


https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/1168ca8e-1bf6-4783-86f9-e80bef616fd5


https://github.com/BeeStation/BeeStation-Hornet/assets/40302913/fe707112-a60b-4ece-9ecb-cb7a656ea7ea



Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
add: thermal mechanics for tcomms
fix: thermal mechanics for RnD servers
fix: upgrading RnD servers
tweak: upgrading RnD servers now makes it use less power and produce less heat
/:cl:
